server/search/search.controller.ts

```diff
@@ -36,35 +36,44 @@ export class SearchController {
         @Res() res: Response,
         @Query("searchText") searchText: string,
         @Query("pageSize") pageSize: number,
-        @Query("page") page: number
+        @Query("page") page: number,
+        @Query("filter") filter: string | Array<string>
     ) {
         try {
             const parsedUrl = parse(req.url, true);
             let searchResults;
-            if (this.configService.get<string>("db.atlas") && searchText) {
+            if (this.configService.get<string>("db.atlas")) {
                 //This is necessary cause the page and pageSize logic is applied to 3 different collections
                 const processedPageSize = Math.ceil((pageSize || 10) / 3);
                 const skipedDocuments =
                     page > 1 ? processedPageSize * (page - 1) : 0;
 
                 const [personalities, sentences, claims] = await Promise.all([
-                    this.personalityService.findAll(
-                        searchText,
-                        processedPageSize,
-                        req.language,
-                        skipedDocuments
-                    ),
-                    this.sentenceService.findAll(
-                        searchText,
-                        processedPageSize,
-                        skipedDocuments
-                    ),
-                    this.claimRevisionService.findAll(
-                        searchText,
-                        processedPageSize,
-                        skipedDocuments
-                    ),
+                    searchText
```

Comment:
suggestion: convert each of these promises into a separate function and use async/await